app/controllers/posts_controller.rb

```diff
@@ -6,4 +6,18 @@ def index
   def show
     @post = Post.find(params[:id])
   end
+
+  def new
+    @post = Post.new
+  end
+
+  def create
+    @post = current_user.posts.build(params.require(:post).permit(:title, :text))
+    if @post.save
+      flash[:success] = 'Post created successfully'
+      redirect_to user_posts_path(id: current_user)
+    else
+      render :new
+    end
+  end
```

Comment:
-  You have done great so far :+1: But there are a few requirements you still have to meet before moving to the next milestone. :+1: :100: 
- [ ] Looking at the video below, you can see that I can't create a post. after filling in the `title and text` and clicking the create button, the transaction starts but gets `ROLLEDBACK` at some point. I'll suggest you look into this as creating a post and eventually a comment should be possible. :+1: :wave: 

- - Video reference:


https://github.com/H-Ramalan/blog/assets/107632780/8b6429bb-bcb8-4ec6-9241-bb781efd8dce


- [OPTIONAL] I'll suggest extracting the post_params into a private method that you can use in the create method. I.e:
```ruby

    private 
    def post_params
      params.require(:post).permit(:title, :text)
    end
```